force-app/main/default/lwc/newProgramEngagement/newProgramEngagement.js

```diff
@@ -69,44 +81,64 @@ export default class NewProgramEngagement extends LightningElement {
         cancelAndBack,
         cantFindContact,
     };
-    engagementObjectApiName = PROGRAMENGAGEMENT_OBJECT;
+    engagementObjectApiName = PROGRAM_ENGAGEMENT_OBJECT;
     contactObjectApiName = CONTACT_OBJECT;
 
     @wire(CurrentPageReference) pageRef;
 
-    @wire(getFieldSet, {
-        objectName: PROGRAMENGAGEMENT_OBJECT.objectApiName,
-        fieldSetName: CREATE_PROGRAM_ENGAGEMENT_FIELD_SET,
+    @wire(getNewEngagementSetup, {
+        programId: "$selectedProgramId",
     })
-    wiredEngagementFields({ error, data }) {
+    wiredSetup({ error, data }) {
         if (data) {
-            this.engagementFieldSet = data;
+            this.cohorts = data.programCohorts;
+            this.loadFieldSets(data);
+            this.setCohortOptions();
         } else if (error) {
             handleError(error);
         }
     }
 
-    @wire(getFieldSet, {
-        objectName: CONTACT_OBJECT.objectApiName,
-        fieldSetName: CREATE_CONTACT_FIELD_SET,
-    })
-    wiredContactFields({ error, data }) {
+    loadFieldSets(data) {
+        if (this.engagementFieldSet && this.contactFieldSet) {
+            return;
+        }
+
+        this.engagementFieldSet = data.engagementFieldSet;
+        this.contactFieldSet = data.contactFieldSet;
+    }
+
+    @wire(getObjectInfo, { objectApiName: PROGRAM_ENGAGEMENT_OBJECT })
+    wiredEngagementInfo({ error, data }) {
         if (data) {
-            this.contactFieldSet = data;
-        } else if (error) {
-            handleError(error);
+            this.recordTypeId = data.defaultRecordTypeId;
+        }
+        if (error) {
+            console.log(error);
         }
     }
 
-    @wire(getProgramCohortsFromProgramId, {
-        programId: "$selectedProgramId",
+    @wire(getPicklistValues, {
+        recordTypeId: "$recordTypeId",
+        fieldApiName: STAGE_FIELD,
     })
-    wiredCohorts({ error, data }) {
-        if (data) {
-            this.cohorts = data;
-            this.setCohortOptions();
+    wiredStageValues({ error, data }) {
+        if (data && data.values) {
```

Comment:
@jjbennett, Do we also need to check if data.defaultValue is there. I am seeing an error when trying to create new PE from BSDT using individual option. I am seeing an empty modal

<img width="1654" alt="Screen Shot 2021-09-07 at 3 05 42 PM" src="https://user-images.githubusercontent.com/34690264/132398352-6a5eda93-fa12-444e-aa8d-56cca735c4f9.png">

Reply:
yes yes we do, thanks!